### PR TITLE
Build .deb file for multiple linux distributions with Docker.

### PR DESCRIPTION
**Pull request by kumagi**
Automatically generate .deb files for 
- `debian-buster` (Debian 10)
- `debian-stretch` (Debian 9)
- `ubuntu-disco` (Ubuntu 19.04)
- `ubuntu-bionic` (Ubuntu 18.04 LTS)
Patterns.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/build
+/ubuntu-*
+/debian-*
```
Comment (yuizumi) [inline .gitignore]: Add end-of-line marker.

File: packaging/build_multiversion_debs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (yuizumi) [inline packaging/build_multiversion_debs]: `build_multiversion_debs` sounds like the deb packages themselves are multi-versioned. I suggest `build_all_debs`.

File: packaging/build_multiversion_debs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# This script must be run at the root directory of the libsxg source tree.
+# Generated deb packages will put in the directory with the same name of
+# Dockerfiles.
+
+
+set -ex
+readonly WORKDIR="${PWD}"
+
+for dockerfile in packaging/dockerfiles/*.dockerfile; do
+  FILENAME="$(basename $dockerfile)"
```
Comment (yuizumi) [inline packaging/build_multiversion_debs]: You can write `$(basename $dockerfile .dockerfile)`.

File: packaging/dockerfiles/debian-buster.dockerfile [hunk 4]
```diff
@@ -0,0 +1,23 @@
+FROM debian:buster-slim
+LABEL maintainer "Hiroki Kumazaki <kumagi@google.com>"
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends -q \
```
Comment (yuizumi) [inline packaging/dockerfiles/debian-buster.dockerfile]: I'm not familiar with docker at all, but can you avoid writing these packages in every dockerfile? Obvious is to create a virtual debian package to install all of these and an apt repository that provides (only) that package, but I know it's way too much.

File: packaging/build_all_debs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  echo "Target: ${distro}"
+  image_tag="${distro}-libsxg"
+  docker build --build-arg base_image=${distro} -t "${image_tag}" -f "${DEB_DOCKER}" .
+  mkdir -p "${distro}"
+  docker run --rm \
+    --mount "type=bind,source=${PWD}/${distro},target=/libsxg/docker_output" \
+    "${image_tag}"
+  echo "Wrote deb file in ${distro}"
+done
+
```
Comment (yuizumi) [inline packaging/build_all_debs]: Remove this extra line.

File: packaging/build_all_debs [hunk 6]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+#
+# Builds the libsxg deb packages for multiple debian family OS with Docker.
+# This script must be run at the root directory of the libsxg source tree.
+# Generated deb packages will put in the directory with the same name of
+# Dockerfiles.
+
+set -ex
+readonly DEB_DOCKER="packaging/dockerfiles/deb.dockerfile"
+declare -ar DISTROS=(
```
Comment (yuizumi) [inline packaging/build_all_debs]: Just using `readonly` might be better in favor of consistency:

```bash
readonly DISTROS=(
  ...
)
```

(I personally write everything with `declare` instead, but it's a matter of taste and I'm fine with `readonly`.)

File: packaging/build_all_debs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+for distro in ${DISTROS[@]}; do
+  echo "Target: ${distro}"
+  image_tag="${distro}-libsxg"
+  docker build --build-arg base_image=${distro} -t "${image_tag}" -f "${DEB_DOCKER}" .
+  mkdir -p "${distro}"
+  docker run --rm \
+    --mount "type=bind,source=${PWD}/${distro},target=/libsxg/docker_output" \
+    "${image_tag}"
+  echo "Wrote deb file in ${distro}"
```
Comment (yuizumi) [inline packaging/build_all_debs]: Period?

File: packaging/dockerfiles/deb.dockerfile [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+                    git \
+                    libssl-dev \
+                    lsb-release && \
+    rm -rf /var/lib/apt/lists/*
+
+ADD . /libsxg
+WORKDIR /libsxg
+
+ENTRYPOINT ["packaging/build_deb", "docker_output"]
+
```
Comment (yuizumi) [inline packaging/dockerfiles/deb.dockerfile]: Same: no empty line at the end of file (unless there's a reason).

File: packaging/build_all_debs (hunk 5 above)
Comment (kumagi) [reply packaging/build_all_debs]: done.

File: packaging/build_all_debs (hunk 6 above)
Comment (kumagi) [reply packaging/build_all_debs]: done.

File: packaging/build_all_debs (hunk 7 above)
Comment (kumagi) [reply packaging/build_all_debs]: appended.

File: packaging/dockerfiles/deb.dockerfile (hunk 8 above)
Comment (kumagi) [reply packaging/dockerfiles/deb.dockerfile]: done.

**Pull request merged**